Analyze/fix test_bgp_update_timer

Test failure seen on backend T1

Version: 20201231.06

Regression seen on the nightly:
```
common_setup_teardown = (<tests.common.helpers.bgp.BGPNeighbor object at 0x7fe850a3d610>, <tests.common.helpers.bgp.BGPNeighbor object at 0x7fe850a3d950>), constants = <test_bgp_update_timer._C object at 0x7fe850a3d390>
duthosts = <tests.common.devices.duthosts.DutHosts object at 0x7fe862a67e10>, rand_one_dut_hostname = 'str2-7050qx-32s-acs-02', toggle_all_simulator_ports_to_rand_selected_tor = None

    def test_bgp_update_timer(common_setup_teardown, constants, duthosts, rand_one_dut_hostname,
                              toggle_all_simulator_ports_to_rand_selected_tor):

        def bgp_update_packets(pcap_file):
            """Get bgp update packets from pcap file."""
            packets = sniff(
                offline=pcap_file,
                lfilter=lambda p: IP in p and bgp.BGPHeader in p and p[bgp.BGPHeader].type == 2
            )
            return packets

        def match_bgp_update(packet, src_ip, dst_ip, action, route):
            """Check if the bgp update packet matches."""
            if not (packet[IP].src == src_ip and packet[IP].dst == dst_ip):
                return False
            subnet = ipaddress.ip_network(route["prefix"].decode())
            _route = (subnet.prefixlen, str(subnet.network_address))
            bgp_fields = packet[bgp.BGPUpdate].fields
            if action == "announce":
                return bgp_fields["tp_len"] > 0 and _route in bgp_fields["nlri"]
            elif action == "withdraw":
                return bgp_fields["withdrawn_len"] > 0 and _route in bgp_fields["withdrawn"]
            else:
                return False

        duthost = duthosts[rand_one_dut_hostname]

        n0, n1 = common_setup_teardown
        try:
            n0.start_session()
            n1.start_session()

            # sleep till new sessions are steady
            time.sleep(30)

            # ensure new sessions are ready
            bgp_facts = duthost.bgp_facts()["ansible_facts"]
            assert n0.ip in bgp_facts["bgp_neighbors"]
            assert n1.ip in bgp_facts["bgp_neighbors"]
            assert bgp_facts["bgp_neighbors"][n0.ip]["state"] == "established"
            assert bgp_facts["bgp_neighbors"][n1.ip]["state"] == "established"

            announce_intervals = []
            withdraw_intervals = []
            for i, route in enumerate(constants.routes):
                bgp_pcap = BGP_LOG_TMPL % i
                with log_bgp_updates(duthost, "any", bgp_pcap):
                    n0.announce_route(route)
                    time.sleep(constants.sleep_interval)
                    n0.withdraw_route(route)
                    time.sleep(constants.sleep_interval)

                with tempfile.NamedTemporaryFile() as tmp_pcap:
                    duthost.fetch(src=bgp_pcap, dest=tmp_pcap.name, flat=True)
                    bgp_updates = bgp_update_packets(tmp_pcap.name)

                announce_from_n0_to_dut = []
                announce_from_dut_to_n1 = []
                withdraw_from_n0_to_dut = []
                withdraw_from_dut_to_n1 = []
                for bgp_update in bgp_updates:
                    if match_bgp_update(bgp_update, n0.ip, n0.peer_ip, "announce", route):
                        announce_from_n0_to_dut.append(bgp_update)
                        continue
                    if match_bgp_update(bgp_update, n1.peer_ip, n1.ip, "announce", route):
                        announce_from_dut_to_n1.append(bgp_update)
                        continue
                    if match_bgp_update(bgp_update, n0.ip, n0.peer_ip, "withdraw", route):
                        withdraw_from_n0_to_dut.append(bgp_update)
                        continue
                    if match_bgp_update(bgp_update, n1.peer_ip, n1.ip, "withdraw", route):
                        withdraw_from_dut_to_n1.append(bgp_update)

                err_msg = "no bgp update %s route %s from %s to %s"
                no_update = False
                if not announce_from_n0_to_dut:
                    err_msg %= ("announce", route, n0.ip, n0.peer_ip)
                    no_update = True
                elif not announce_from_dut_to_n1:
                    err_msg %= ("announce", route, n1.peer_ip, n1.ip)
                    no_update = True
                elif not withdraw_from_n0_to_dut:
                    err_msg %= ("withdraw", route, n0.ip, n0.peer_ip)
                    no_update = True
                elif not withdraw_from_dut_to_n1:
                    err_msg %= ("withdraw", route, n1.peer_ip, n1.ip)
                    no_update = True
                if no_update:
>                   pytest.fail(err_msg)
E                   Failed: no bgp update announce route {'prefix': '10.10.100.0/27', 'nexthop': '192.168.0.2'} from 192.168.0.1 to 192.168.0.3

bgp/test_bgp_update_timer.py:222: Failed
```

--- a/tests/bgp/test_bgp_update_timer.py
+++ b/tests/bgp/test_bgp_update_timer.py
@@ -72,7 +72,7 @@ def common_setup_teardown(duthosts, rand_one_dut_hostname, is_dualtor, is_quagga
         if k == duthost.hostname:
             dut_type = v['type']
 
-    if dut_type == 'ToRRouter':
+    if 'ToRRouter' in dut_type:
         neigh_type = 'LeafRouter'
     else:
         neigh_type = 'ToRRouter'

